engine.py

```diff
@@ -197,6 +211,153 @@ def log_error(self, msg):
         
     
     ##########################################################################################
+    # panel interfaces
+    
+    def _generate_panel_id(self, dialog_name, bundle):
+        """
+        Given a dialog name and a bundle, generate a Nuke panel id.
+        This panel id is used by Nuke to identify and persist the panel.
+        
+        This will return something like 'shotgun_tk_multi_loader2_main'
+        
+        :param dialog_name: An identifier string to identify the dialog to be hosted by the panel
+        :param bundle: The bundle (e.g. app) object to be associated with the panel
+        :returns: a unique identifier string 
+        """
+        panel_id = "shotgun_%s_%s" % (bundle.name, dialog_name)
+        # replace any non-alphanumeric chars with underscores
+        panel_id = re.sub("\W", "_", panel_id)
+        panel_id = panel_id.lower()
+        self.log_debug("Unique panel id for %s %s -> %s" % (bundle, dialog_name, panel_id))
+        return panel_id 
+        
+    def _panel_factory_callback(self, panel_id):
+        """
+        Given a panel id, generate a panel UI.
+        
+        This method is intended to be executed by nuke itself as a callback
```

Comment:
I am not totally sure I totally understand your comment here. Nevertheless, this nuke specific panel restore callback is now gone in the reshuffled panel structure.